applications/StructuralMechanicsApplication/custom_constitutive/truss_plasticity_constitutive_law.cpp

```diff
@@ -0,0 +1,294 @@
+    const Variable<bool>& rThisVariable,
+    bool& rValue
+    )
+{
+    if(rThisVariable == INELASTIC_FLAG) rValue = this->mInelasticFlag;
+    else KRATOS_ERROR << "Can't get the specified value" << std::endl;
+    return rValue;
+}
+
+//************************************************************************************
+//************************************************************************************
+
+double& TrussPlasticityConstitutiveLaw::GetValue(
+    const Variable<double>& rThisVariable,
+    double& rValue
+    )
+{
+    if(rThisVariable == PLASTIC_STRAIN) rValue = this->mAccumulatedPlasticStrain;
+    else if(rThisVariable == PLASTIC_ALPHA) rValue = this->mPlasticAlpha;
+    else KRATOS_ERROR << "Can't get the specified value" << std::endl;
+    return rValue;
+}
+
+//************************************************************************************
+//************************************************************************************
+
+array_1d<double, 3 > & TrussPlasticityConstitutiveLaw::GetValue(
+    const Variable<array_1d<double, 3 > >& rThisVariable,
+    array_1d<double, 3 > & rValue)
+{
+    if (rThisVariable == FORCE)
+    {
+        rValue[0] = this->mStressState;
+        rValue[1] = 0.0;
+        rValue[2] = 0.0;
+    }
+    else KRATOS_ERROR << "Can't get the specified value" << std::endl;
+    return rValue;
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void TrussPlasticityConstitutiveLaw::SetValue(
+    const Variable<double>& rThisVariable,
+    const double& rValue,
+    const ProcessInfo& rCurrentProcessInfo
+    )
+{
+    if(rThisVariable == PLASTIC_STRAIN) this->mAccumulatedPlasticStrain = rValue;
+    if(rThisVariable == PLASTIC_ALPHA) this->mPlasticAlpha = rValue;
+    else KRATOS_ERROR << "Can't set the specified value" << std::endl;
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void TrussPlasticityConstitutiveLaw::SetValue(const Variable<bool>& rThisVariable,
+                          const bool& rValue,
+                          const ProcessInfo& rCurrentProcessInfo)
+{
+    if(rThisVariable == INELASTIC_FLAG) this->mInelasticFlag = rValue;
+    else KRATOS_ERROR << "Can't set the specified value" << std::endl;
+}
+
+
+//************************************************************************************
+//************************************************************************************
+
+double& TrussPlasticityConstitutiveLaw::CalculateValue(
+    ConstitutiveLaw::Parameters& rParameterValues,
+    const Variable<double>& rThisVariable,
+    double& rValue
+    )
+{
+    if(rThisVariable == TANGENT_MODULUS)
+    {    
+        const Properties& r_material_properties = rParameterValues.GetMaterialProperties(); 
```

Comment:
if you use `auto` here then you don't have to include `properties.h`

Reply:
CalculateMaterialResponse has a Properties as an argument and is a base function. So I guess I need properties.h anyway

Reply:
Hm if properties.h is included in the baseclass then you dont need the include here I would say